From f82e13b8e955717ae21a5f002f6cc054301abcb8 Mon Sep 17 00:00:00 2001
From: Iain Lane <laney@ubuntu.com>
Date: Sun, 9 Jan 2011 23:22:12 +0000
Subject: [PATCH] Add patch to fix FTBFS with --no-add-needed

debian/patches/linking-no-add-needed.patch: Add new patch to
explicitly link with the math libraries, to resolve a FTBFS with
binutils-gold/--no-add-needed. (Closes: #556302)
---
 debian/changelog                           |    8 +++++++
 debian/patches/linking-no-add-needed.patch |   31 ++++++++++++++++++++++++++++
 debian/patches/series                      |    1 +
 3 files changed, 40 insertions(+), 0 deletions(-)
 create mode 100644 debian/patches/linking-no-add-needed.patch

diff --git a/debian/changelog b/debian/changelog
index ef0f784..5aba8f5 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,11 @@
+qhull (2009.1-2) UNRELEASED; urgency=low
+
+  * debian/patches/linking-no-add-needed.patch: Add new patch to
+    explicitly link with the math libraries, to resolve a FTBFS with
+    binutils-gold/--no-add-needed. (Closes: #556302)
+
+ -- Iain Lane <laney@ubuntu.com>  Sun, 09 Jan 2011 23:20:00 +0000
+
 qhull (2009.1-1) unstable; urgency=low
 
   * New upstream release
diff --git a/debian/patches/linking-no-add-needed.patch b/debian/patches/linking-no-add-needed.patch
new file mode 100644
index 0000000..c96bc75
--- /dev/null
+++ b/debian/patches/linking-no-add-needed.patch
@@ -0,0 +1,31 @@
+Explicitly link with -lm for the math libraries to be available to the linker,
+required to fix FTBFS with binutils-gold/--no-add-needed
+
+  -- Iain Lane <laney@ubuntu.com>  Sun, 09 Jan 2011 23:19:48 +0000
+
+Index: qhull-2009.1/src/Makefile.am
+===================================================================
+--- qhull-2009.1.orig/src/Makefile.am	2011-01-09 23:04:26.854227305 +0000
++++ qhull-2009.1/src/Makefile.am	2011-01-09 23:12:38.512198225 +0000
+@@ -40,7 +40,7 @@
+ 
+ # how:
+ qhull_LDADD = libqhull.la
+-rbox_LDADD = libqhull.la
++rbox_LDADD = libqhull.la -lm
+ qconvex_LDADD = libqhull.la
+ qdelaunay_LDADD = libqhull.la
+ qvoronoi_LDADD = libqhull.la
+Index: qhull-2009.1/src/Makefile.in
+===================================================================
+--- qhull-2009.1.orig/src/Makefile.in	2011-01-09 23:12:43.582589854 +0000
++++ qhull-2009.1/src/Makefile.in	2011-01-09 23:12:51.993239477 +0000
+@@ -263,7 +263,7 @@
+ 
+ # how:
+ qhull_LDADD = libqhull.la
+-rbox_LDADD = libqhull.la
++rbox_LDADD = libqhull.la -lm
+ qconvex_LDADD = libqhull.la
+ qdelaunay_LDADD = libqhull.la
+ qvoronoi_LDADD = libqhull.la
diff --git a/debian/patches/series b/debian/patches/series
index 1b7b72a..595b1ee 100644
--- a/debian/patches/series
+++ b/debian/patches/series
@@ -1,2 +1,3 @@
 make-new-msg.patch
 QHpointer.patch
+linking-no-add-needed.patch
-- 
1.7.2.3

